### PR TITLE
Initial code for snmp polling of port status.

### PR DESCRIPTION
**Pull request by koteswar16**
Initial code for snmp polling of port status.

### CODE REVIEW

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py [hunk 1]
```diff
@@ -56,11 +58,48 @@ class HPSNMPProvisioningDriver(api.NetworkProvisioningApi):
...
+                        db.update_bnp_phys_swport_status(
+                            self.context, swport['switch_id'],
+                            swport['interface_name'], status)
+            eventlet.sleep(self.bnp_sync_interval)
+
+    def start_snmp_polling(self):
+        """Spawn a thread to poll the switch db."""
+        eventlet.spawn(self._snmp_sync_thread)
+
     def create_port(self, port):
```
Comment (hp-networking) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: This is old file. please resync your repo with master branch

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py [hunk 2]
```diff
@@ -56,11 +58,48 @@ class HPSNMPProvisioningDriver(api.NetworkProvisioningApi):
...
+            self.start_snmp_polling()
         # TODO(selva) need to check how we can load dynamically
         drvr = 'baremetal_network_provisioning.drivers.snmp_driver.SNMPDriver'
         self.context = neutron_context.get_admin_context()
         self._load_drivers(drvr)
 
+    def _snmp_sync_thread(self):
+        """Sync switch database periodically."""
+        LOG.info(_LI('BMNP start thread....'))
+        while True:
```
Comment (hp-networking) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: As discussed read the syn_enable flag.

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py [hunk 3]
```diff
@@ -56,11 +58,48 @@ class HPSNMPProvisioningDriver(api.NetworkProvisioningApi):
...
+            portmaps = db.get_all_bnp_swport_mappings(self.context)
+            for portmap in portmaps:
+                swport = db.get_bnp_phys_switch_port_by_id(
+                    self.context, portmap['switch_port_id'])
+                switch = db.get_bnp_phys_switch(self.context,
+                                                portmap['switch_id'])
+                try:
+                    snmp_drv = discovery_driver.SNMPDiscoveryDriver(switch)
```
Comment (hp-networking) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: This code is not working please resolve the below error ,
 SNMP operation 'GET' failed: NoSuchObjectError({'str': "Can't resolve node name ::(u'1', u'3', u'6', u'1', u'2', u'1', u'2', u'2', u'1', u'8', u'12') at <pysnmp.smi.view.MibViewController instance at 0x7f1e0c2dad40>"})^[[00m

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py (hunk 2 above)
Comment (koteswar16) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: i am doing this at line no: 65. ie before entering into this function.

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py (hunk 3 above)
Comment (koteswar16) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: i will check.

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py (hunk 1 above)
Comment (koteswar16) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: Already up-to-date. There are two functions get_bnp_phys_port_by_id and get_bnp_phys_switch_port_by_id which are doing same. So I deleted get_bnp_phys_port_by_id and used get_bnp_phys_switch_port_by_id  every where.

File: baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py (hunk 3 above)
Comment (koteswar16) [inline baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py]: I cloned my forked repo and installed and created port. it didn't give any error. please check once.

**Pull request merged**